test/dates/periods.jl

```diff
@@ -292,6 +322,15 @@ dt = Dates.DateTime(2014)
 @test Dates.toms(d)  == Dates.value(Dates.Millisecond(d)) == 86400000
 @test Dates.toms(w)  == Dates.value(Dates.Millisecond(w)) == 604800000
 
+@test Dates.tons(ns) == Dates.value(Dates.Nanosecond(ns)) == 1
+@test Dates.tons(us) == Dates.value(Dates.Nanosecond(us)) == 1000
+@test Dates.tons(ms) == Dates.value(Dates.Nanosecond(ms)) == 1000000
+@test Dates.tons(s)  == Dates.value(Dates.Nanosecond(s))  == 1000000000
+@test Dates.tons(mi) == Dates.value(Dates.Nanosecond(mi)) == Int64(60000000000)
```

Comment:
@quinnj I think I found the issue `N = 1` should be `N = Int64(1)`. See https://github.com/JuliaLang/julia/pull/12274/files#diff-18ff5efda613bb05a8b004a550205c03R402